cache unusable after version update

**Actual behavior**
When cache entries were created prior to v1.25.0, they now can't be consumed by v1.25.0 and result in an error
```
INFO[0012] Found cached layer, extracting to filesystem 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x191e5d5]
goroutine 1 [running, locked to thread]:
github.com/GoogleContainerTools/kaniko/pkg/executor.(*stageBuilder).convertLayerMediaType(0xc000531ba0, {0x0, 0x0})
	/src/pkg/executor/build.go:595 +0x35
github.com/GoogleContainerTools/kaniko/pkg/executor.(*stageBuilder).saveLayerToImage(0xc000531ba0, {0x0?, 0x0?}, {0xc0005025a0, 0x11})
	/src/pkg/executor/build.go:630 +0x47
github.com/GoogleContainerTools/kaniko/pkg/executor.(*stageBuilder).build(0xc000531ba0)
	/src/pkg/executor/build.go:429 +0xfef
github.com/GoogleContainerTools/kaniko/pkg/executor.DoBuild(0x33b0be0)
	/src/pkg/executor/build.go:777 +0x5ea
github.com/GoogleContainerTools/kaniko/cmd/executor/cmd.init.func2(0xc0001dc900?, {0x1fa74a0?, 0x4?, 0x1fa74a4?})
	/src/cmd/executor/cmd/root.go:190 +0x1a7
github.com/spf13/cobra.(*Command).execute(0x3399b80, {0xc00004e130, 0x10, 0x11})
	/src/vendor/github.com/spf13/cobra/command.go:1019 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x3399b80)
	/src/vendor/github.com/spf13/cobra/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(0x1fb2dcd?)
	/src/vendor/github.com/spf13/cobra/command.go:1071 +0x13
main.main()
	/src/cmd/executor/main.go:31 +0x5b
```

**Expected behavior**

The outdated cache gets rejected and a new layer is built instead, cache is overwritten with the new version. Or the cache includes a version number, s.t. the incompatible cache is not pulled in the first place. At the minimum the pointer should be checked and a proper error message returned, not segfault.

**To Reproduce**

1. build image with v1.24.3, push to cache
2. rebuild image with v1.25.0

**Additional Information**

This happens for layers that triggered special behaviour in <v1.25.0 where we mimicked legacy docker behaviour, specifically empty RUN statements, ie `RUN echo hello` and empty WORKDIR, ie. `WORKDIR /`

We redefined their behaviour when we switched to mimicking buildkit 
https://github.com/mzihlmann/kaniko/pull/81/files#diff-815db7a7d24db4a24dfcaa06524762a4d2631187778eff8cdc6834a50532f493L429
Now those cases always do emit layers and expect them to be there too.

--- a/pkg/executor/build.go
+++ b/pkg/executor/build.go
@@ -426,8 +426,14 @@ func (s *stageBuilder) build() error {
 		if isCacheCommand {
 			v := command.(commands.Cached)
 			layer := v.Layer()
-			if err := s.saveLayerToImage(layer, command.String()); err != nil {
-				return errors.Wrap(err, "failed to save layer")
+			if layer == nil {
+				// a cache image without a layer indicates that no files were changed, ie. by 'WORKDIR /' prior to v1.25.0
+				// We continue to handle this case here as users might still have cache entries lying around
+				logrus.Info("No files were changed, appending empty layer to config. No layer added to image.")
+			} else {
+				if err := s.saveLayerToImage(layer, command.String()); err != nil {
+					return errors.Wrap(err, "failed to save layer")
+				}
 			}
 		} else {
 			tarPath, err := s.takeSnapshot(files, command.ShouldDetectDeletedFiles())

